foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_7_0/SparkAppExecMigration.java

```diff
@@ -0,0 +1,91 @@
+ * use.
+ */
+public class SparkAppExecMigration extends AbstractAissembleMigration {
+
+    public static final String NEW_CHART = "${aissemble.helm.repo.protocol}://${aissemble.helm.repo}/aissemble-spark-application-chart";
+
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        Model pom = PomHelper.getLocationAnnotatedModel(file);
+        return !getSparkAppExecutions(pom).isEmpty();
+    }
+
+    @Override
+    protected boolean performMigration(File file) {
+        Model pom = PomHelper.getLocationAnnotatedModel(file);
+        List<PluginExecution> execs = getSparkAppExecutions(pom);
+        PomModifications modifications = new PomModifications();
+        for (PluginExecution exec : execs) {
+            modifyExecution(exec, modifications);
+        }
+        return PomHelper.writeModifications(file, modifications.finalizeMods());
+    }
+
+    private void modifyExecution(PluginExecution exec, PomModifications modifications) {
+        Xpp3Dom[] args = getConfig(exec, "arguments").getChildren("argument");
+        for (Xpp3Dom arg : args) {
+            InputLocation location = (InputLocation) arg.getInputLocation();
+            String value = arg.getValue();
+            if ("aissemble-spark-application".equals(value)) {
+                InputLocation end = PomHelper.incrementColumn(location, "aissemble-spark-application".length());
+                modifications.add(new PomModifications.Replacement(location, end, NEW_CHART));
+            } else if ("--repo".equals(value) || "${aissemble.helm.repo}".equals(value)) {
+                InputLocation start = PomHelper.incrementColumn(location, -"<argument>".length());
+                InputLocation end = PomHelper.incrementColumn(location, value.length() + "</argument>".length());
+                modifications.add(new PomModifications.Deletion(start, end));
+            }
+        }
+    }
+
+    private List<PluginExecution> getSparkAppExecutions(Model pom) {
+        return pom.getBuild().getPlugins().stream().filter(plugin -> plugin.getArtifactId().equals("exec-maven-plugin"))
+                .flatMap(plugin -> plugin.getExecutions().stream())
+                .filter(execution -> execution.getGoals().contains("exec"))
+                .filter(this::isHelmExec)
+                .filter(this::containsOldSparkAppArg)
+                .collect(Collectors.toList());
+    }
+
+    private boolean isHelmExec(PluginExecution execution) {
+        return getConfig(execution, "executable").getValue().equals("helm");
+    }
+
+    private boolean containsOldSparkAppArg(PluginExecution execution) {
+        Xpp3Dom args = ((Xpp3Dom) execution.getConfiguration()).getChild("arguments");
+        return Stream.of(args.getChildren("argument"))
+                .anyMatch(arg -> "aissemble-spark-application".equals(arg.getValue()));
+    }
+
+    private Xpp3Dom getConfig(PluginExecution execution, String name) {
```

Comment:
A: Javadocs on this file

Reply:
I think we got the class-level comment, which is the major compulsory one here.  Are there specific items you would like to see expanded upon more?